pkg/kubectl/cmd/cmd_printing_test.go

```diff
@@ -0,0 +1,220 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
+	"k8s.io/kubernetes/pkg/printers"
+)
+
+func TestIllegalPackageSourceCheckerThroughPrintFlags(t *testing.T) {
+	testCases := []struct {
+		name                 string
+		expectInternalObjErr bool
+		output               string
+		obj                  runtime.Object
+		expectedOutput       string
+	}{
+		{
+			name:                 "success printer: object containing package path beginning with forbidden prefix is rejected",
+			expectInternalObjErr: true,
+			obj:                  internalPod(),
+		},
+		{
+			name:                 "success printer: object containing package path with no forbidden prefix returns no error",
+			expectInternalObjErr: false,
+			obj:                  externalPod(),
+			output:               "",
+			expectedOutput:       "pod/foo succeeded\n",
+		},
+		{
+			name:                 "name printer: object containing package path beginning with forbidden prefix is rejected",
+			expectInternalObjErr: true,
+			output:               "name",
+			obj:                  internalPod(),
+		},
+		{
+			name:                 "json printer: json printer is wrapped in a versioned printer - internal obj should be converted with no error",
+			expectInternalObjErr: false,
+			output:               "json",
+			obj:                  internalPod(),
+		},
+		{
+			name:                 "yaml printer: yaml printer is wrapped in a versioned printer - internal obj should be converted with no error",
+			expectInternalObjErr: false,
+			output:               "yaml",
+			obj:                  internalPod(),
+		},
+	}
+
+	for _, tc := range testCases {
+		printFlags := printers.NewPrintFlags("succeeded")
+		printFlags.OutputFormat = &tc.output
+
+		printer, err := printFlags.ToPrinter()
+		if err != nil {
+			t.Fatalf("unexpected error %v", err)
+		}
+
+		output := bytes.NewBuffer([]byte{})
+
+		err = printer.PrintObj(tc.obj, output)
+		if err != nil {
+			if !tc.expectInternalObjErr {
+				t.Fatalf("unexpected error %v", err)
+			}
+
+			if !printers.IsInternalObjectError(err) {
+				t.Fatalf("unexpected error - expecting internal object printer error, got %q", err)
+			}
+			continue
+		}
+
+		if tc.expectInternalObjErr {
+			t.Fatalf("expected internal object printer error, but got no error")
+		}
+
+		if len(tc.expectedOutput) == 0 {
+			continue
+		}
+
+		if tc.expectedOutput != output.String() {
+			t.Fatalf("unexpected output: expecting %q, got %q", tc.expectedOutput, output.String())
+		}
+	}
+}
+
+func TestIllegalPackageSourceCheckerDirectlyThroughPrinters(t *testing.T) {
```

Comment:
@deads2k added additional tests to ensure all printers#PrintObj method fails on internal objects